### PR TITLE
part of #1017, split the lib directory

### PR DESCRIPTION
**Pull request by huafengw**

### CODE REVIEW

File: core/src/main/scala/org/apache/gearpump/jarstore/JarFileContainerWrapper.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * limitations under the License.
+ */
+package org.apache.gearpump.jarstore
+
+import java.io._
+
+import org.apache.gearpump.cluster.UserConfig
+import JarFileContainerWrapper._
+
+class JarFileContainerWrapper private (conf: UserConfig) extends JarFileContainer{
```
Comment (manuzhang) [inline core/src/main/scala/org/apache/gearpump/jarstore/JarFileContainerWrapper.scala]: maybe some comments here on why we need this "wrapper"

File: core/src/main/scala/org/apache/gearpump/jarstore/JarFileContainerWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * limitations under the License.
+ */
+package org.apache.gearpump.jarstore
+
+import java.io._
+
+import org.apache.gearpump.cluster.UserConfig
+import JarFileContainerWrapper._
+
+/**
```
Comment (clockfly) [inline core/src/main/scala/org/apache/gearpump/jarstore/JarFileContainerWrapper.scala]: I think the JarFileContainerWrapper itself is clear. No need to add class comments in the class definition of JarFileContainerWrapper.

We only need to explain why we use JarFileContainerWrapper at where it is used. Like JarStoreService.

**Comment by clockfly** [comment]
+1

**Comment by clockfly** [comment]
@huafengw 

Please be 100% sure that this PR won't break existing function before merging.

**Pull request merged**